File: Algebra/Fundamentals/Binary-Exponentiation/BinaryExponentiation.java
public class BinaryExponentiation{
    public long binpow(long a, long n) {
        if (n == 0) {
            return 1;
        }
        long res = binpow(a, n/2);
        if  (n%2 == 1) {
            return res * res * a;
        }
        else {
            return res * res;
        }
    }

    public static void main(String[] args) {
        long a = 3;
        long n = 13;
        BinaryExponentiation res = new BinaryExponentiation();
        long ans = res.binpow(a, n);
        System.out.println("Answer for 3 to the power of 13 is: " + ans);
    }
}

File: Algebra/Fundamentals/Binary-Exponentiation/BinaryExponentiation.py
class BinaryExponentiation:

    def binPow(self, base, power):
        if power == 0:
            return 1
        res = self.binPow(base, power//2)

        if (power % 2):
            return res * res * base
        return res * res
    

binEx = BinaryExponentiation()
res = binEx.binPow(3, 13)
print("The result for 3 to the power of 13 is: ", res)